docs/installation/debian.md

```diff
@@ -12,113 +13,121 @@ parent = "smn_linux"
+ display your kernel version:
 
-To install the latest Debian package (may not be the latest Docker release):
+     $ uname -r
 
-    $ sudo apt-get update
-    $ sudo apt-get install docker.io
+### Update your apt repository
 
-To verify that everything has worked as expected:
+Docker's `apt` repository contains Docker 1.7.1 and higher. To set `apt` to use from the new repository:
 
-    $ sudo docker run --rm hello-world
+ 1. If you haven't already done so, log into your Ubuntu instance.
 
-This command downloads and runs the `hello-world` image in a container. When the
-container runs, it prints an informational message. Then, it exits.
+ 2. Open a terminal window.
 
-If you need to add an HTTP Proxy, set a different directory or partition for the
-Docker runtime files, or make other customizations, read our Systemd article to
-learn how to [customize your Systemd Docker daemon options](/articles/systemd/).
+ 3. Add the new `gpg` key.
 
-> **Note**:
-> If you want to enable memory and swap accounting see
-> [this](/installation/ubuntulinux/#adjust-memory-and-swap-accounting).
+         $ apt-key adv --keyserver hkp://pgp.mit.edu:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
 
-### Uninstallation
+ 4. Open the `/etc/apt/sources.list.d/docker.list` file in your favorite editor.
 
-To uninstall the Docker package:
+     If the file doesn't exist, create it.
 
-    $ sudo apt-get purge docker.io
+ 5. Remove any existing entries.
 
-To uninstall the Docker package and dependencies that are no longer needed:
+ 6. Add an entry for your Debian operating system.
 
-    $ sudo apt-get autoremove --purge docker.io
+     The possible entries are:
 
-The above commands will not remove images, containers, volumes, or user created
-configuration files on your host. If you wish to delete all images, containers,
-and volumes run the following command:
+       # Debian Wheezy
+      deb https://apt.dockerproject.org/repo debian-wheezy main
 
-    $ rm -rf /var/lib/docker
+      # Debian Jessie
+      deb https://apt.dockerproject.org/repo debian-jessie main
 
-You must delete the user created configuration files manually.
+      # Debian Stretch/Sid
+      deb https://apt.dockerproject.org/repo debian-stretch main
 
-## Debian Wheezy/Stable 7.x (64-bit)
+ 7. Save and close the file.
 
-Docker requires Kernel 3.8+, while Wheezy ships with Kernel 3.2 (for more details
-on why 3.8 is required, see discussion on
-[bug #407](https://github.com/docker/docker/issues/407)).
+ 8. Update the `apt` program.
 
-Fortunately, wheezy-backports currently has [Kernel 3.16
-](https://packages.debian.org/search?suite=wheezy-backports&section=all&arch=any&searchon=names&keywords=linux-image-amd64),
-which is officially supported by Docker.
+         $ apt-get update
 
-### Installation
+ 9. Purge the old repo if it exists.
 
-1. Install Kernel from wheezy-backports
+         $ apt-get purge lxc-docker*
```

Comment:
I think we should move this to the "install docker" section, because this will actually remove the previous install, not just the repo. To minimize downtime, might be better to put purging the previous install, and installing the new one close to each other. Wdyt?